streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -265,7 +265,7 @@ private void verifyAllKVKeys(final StreamRunnable[] streamRunnables, final Kafka
                 public boolean conditionMet() {
                     try {
                         final StreamsMetadata metadata = streams.metadataForKey(storeName, key, new StringSerializer());
-                        if (metadata == null) {
+                        if (metadata == null || metadata.equals(StreamsMetadata.NOT_AVAILABLE)) {
```

Comment:
`Optional<StreamsMetadata>` would be a reasonable solution for the reinitialisation case, but we can't use it sadly.